src/Catalog/Registration/RegistrationMakerCatalogItem.cs

```diff
@@ -214,5 +215,25 @@ public override IGraph CreatePageContent(CatalogContext context)
                 throw new Exception(string.Format("Exception processing catalog item {0}", _catalogUri), e);
             }
         }
+
+        private string GetIconUrl()
+        {
+            var subject = _catalogItem.CreateUriNode(_catalogUri);
+            var packageId = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.Id)).FirstOrDefault().Object.ToString();
+            var packageVersion = NuGetVersionUtility.NormalizeVersion(_catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.Version)).FirstOrDefault().Object.ToString());
+            var iconUrlTriple = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.IconUrl)).FirstOrDefault();
+            var iconFileTriple = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.IconFile)).FirstOrDefault();
+
+            if (!string.IsNullOrWhiteSpace(iconFileTriple?.Object.ToString()) && !string.IsNullOrWhiteSpace(packageId) && !string.IsNullOrWhiteSpace(packageVersion))
+            {
+                // The embedded icon file case. We assume here that catalog2dnx did its job
+                // and extracted the icon file to the appropriate location.
+                string path = PackagePathProvider.GetIconPath(packageId, packageVersion);
+                return new Uri(_packageContentBaseAddress, path).AbsoluteUri;
+            }
+
+            // TODO: Update when iconUrl ingestion work is done: https://github.com/nuget/nugetgallery/issues/7061
+            return iconUrlTriple?.Object.ToString() ?? "";
```

Comment:
It is ok to have the empty entries in catalog?

Reply:
This produces an empty string value in registration. And yes, that's existing behavior. See [here](https://api.nuget.org/v3/registration3-gz-semver2/e2e.signedpackage/page/1.0.0-180502.211641.6633165/2018.7.19-v193608.json) for example.